src/pip/_internal/commands/install.py

```diff
@@ -45,10 +45,12 @@
 logger = logging.getLogger(__name__)
 
 
-def get_check_binary_allowed(format_control):
-    # type: (FormatControl) -> BinaryAllowedPredicate
+def get_check_binary_allowed(format_control, features_enabled):
+    # type: (FormatControl, List[str]) -> BinaryAllowedPredicate
     def check_binary_allowed(req):
         # type: (InstallRequirement) -> bool
+        if "always-install-via-wheel" in features_enabled:
+            return True
```

Comment:
Would it be cleaner to have this logic outside of this function, in `run()` instead?

Reply:
I don't think so, because the point is to get a `BinaryAllowedPredicate` that is passed around down to `_should_build`.